OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -2807,6 +2807,13 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				// rename WithDockingOverlay to WithDockedOverlay
+				if (engineVersion < 20160101)
+				{
+					if (node.Key.StartsWith("WithDockingOverlay"))
+						node.Key = "WithDockedOverlay";
```

Comment:
You can append `node.Key.Substring(18)` to retain any `@suffix` that the original had (such as `smoke` in the rules below).